src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -141,7 +148,29 @@ public void InsertCode(Address address, ReadOnlyMemory<byte> code, IReleaseSpec
         {
             _needsStateRootUpdate = true;
             Keccak codeHash = code.Length == 0 ? Keccak.OfAnEmptyString : Keccak.Compute(code.Span);
-            _codeDb[codeHash.Bytes] = code.ToArray();
+
+            // Don't reinsert if already inserted. This can be the case when the same
+            // code is used by multiple deployments. Either from factory contracts (e.g. LPs)
+            // or people copy and pasting popular contracts
+            if (!_codeInsertFilter.Get(codeHash))
```

Comment:
what if a re-orged block inserted something, isn't it a problem (maybe not, as it will be in db)

Reply:
Doesn't matter; hash->code is never removed from db; the codehash on account may be reorg'd but the code won't be deleted (would be too complicated to track as could be thousands of accounts pointing at same code)